PmipCallStackFilter.cs

```diff
@@ -50,70 +56,78 @@ public static DkmStackWalkFrame PmipStackFrame(DkmStackContext stackContext, Dkm
+                    _fileStreamReader = new StreamReader(_fileStream);
+                    _currentFile = fileName;
+                }
+                catch (Exception ex)
                 {
-                    using (var file = new StreamReader(inStream))
-                    {
-                        string line;
-                        while ((line = file.ReadLine()) != null)
-                        {
-                            const char delemiter = ';';
-                            var tokens = line.Split(delemiter);
-
-                            //should never happen, but lets be safe and not get array out of bounds if it does
-                            if (tokens.Length != 3)
-                                continue;
-
-                            var startip = tokens[0];
-                            var endip = tokens[1];
-                            var description = tokens[2];
-
-                            var startiplong = ulong.Parse(startip, NumberStyles.HexNumber);
-                            var endipint = ulong.Parse(endip, NumberStyles.HexNumber);
-
-                            list.Add(new Range() { Name = description, Start = startiplong, End = endipint });
-                        }
-                    }
+                    Console.WriteLine("Unable to read dumped pmip file: " + ex.Message);
                 }
+            }
+
+            try
+            {
+                string line;
+                while ((line = _fileStreamReader.ReadLine()) != null)
+                {
+                    const char delemiter = ';';
+                    var tokens = line.Split(delemiter);
+
+                    //should never happen, but lets be safe and not get array out of bounds if it does
+                    if (tokens.Length != 3)
+                        continue;
+
+                    var startip = tokens[0];
+                    var endip = tokens[1];
+                    var description = tokens[2];
 
-                list.Sort((r1, r2) => r1.Start.CompareTo(r2.Start));
-                _rangesSortedByIp = list.ToArray();
-                _previousFileLength = fileInfo.Length;
+                    var startiplong = ulong.Parse(startip, NumberStyles.HexNumber);
+                    var endipint = ulong.Parse(endip, NumberStyles.HexNumber);
+                    _rangesSortedByIp.Add(new Range() { Name = description, Start = startiplong, End = endipint });
+                }
             }
             catch (Exception ex)
             {
                 Console.WriteLine("Unable to read dumped pmip file: " + ex.Message);
             }
-
         }
 
         public static bool TryGetDescriptionForIp(ulong ip, out string name)
         {
             name = string.Empty;
 
-            if (_rangesSortedByIp == null)
-                return false;
+            if(_count != _rangesSortedByIp.Count)
```

Comment:
Do we want to sort here now, or put it back with the file reading? Then we can drop the `_count` variable I think.

Reply:
Sorting after reading the file is nicer I'd say :)